src/app/components/users-groups/group-users/group-users.component.ts

```diff
@@ -7,73 +7,100 @@ import {DeleteDialogComponent} from '../../dialogs/delete-dialog/delete-dialog.c
 import {MatDialog} from '@angular/material/dialog';
 import {UserService} from '../../../services/user-service/user-service';
 import {SnackBarService} from '../../../services/snack-bar/snack-bar-service';
+import {Realm} from '../../../models/Realm';
+import {RoleService} from '../../../services/role-service/role-service';
+import {Role} from '../../../models/Role';
 
 @Component({
   selector: 'app-group-users',
   templateUrl: './group-users.component.html',
   styleUrls: ['./group-users.component.css']
 })
 export class GroupUsersComponent implements OnInit {
+  realm: Realm;
+  private subscription;
 
   constructor(private router: Router,
               private groupService: GroupService,
               private userService: UserService,
               private dialog: MatDialog,
-              private snackbar: SnackBarService ) {
+              private snackbar: SnackBarService,
+              private roleService: RoleService) {
   }
 
   group: Group;
   users: User[];
+  roles: Role[];
   groupUsers: User[] = [];
 
   ngOnInit(): void {
-    this.getGroup();
+    this.subscription = this.groupService.groups.subscribe(() => {
+      this.getGroupUsers();
+      this.getAllUsers();
+      this.getAllRoles();
+    }, error => this.snackbar.openSnackBar(error.error.message, 4000));
+    this.getGroupUsers();
+    this.getAllRoles();
     this.getAllUsers();
   }
 
-  getGroup() {
-    let group = localStorage.getItem('groupName');
-    this.groupService.getGroupByName(group).subscribe(data => {
-      this.group = data;
-    },error => {
-      this.snackbar.openSnackBar(error.error.message, 3000);
-    });
-    this.groupService.getUsersFromGroup(group).subscribe(data => {
-      this.groupUsers = data;
-    },error => {
-      this.snackbar.openSnackBar(error.error.message, 3000);
+  addRoleToAllUsers(role) {
+    let realm = localStorage.getItem('realm');
+    this.groupService.addRoleToGroup(JSON.parse(realm).name, this.group.name, role.name).subscribe();
+  }
+
+  getAllRoles() {
+    let realm = localStorage.getItem('realm');
+
+    this.roleService.getAllRoles(JSON.parse(realm).name).subscribe(data => {
+      this.roles = data;
     });
   }
 
+  ngOnDestroy() {
+    this.subscription.unsubscribe();
+  }
+
   getAllUsers() {
-    this.userService.getAllUsers().subscribe(data => {
+    let realm = localStorage.getItem('realm');
+
+    this.userService.getAllUsers(JSON.parse(realm).name).subscribe(data => {
       this.users = data;
-    },error => {
-      this.snackbar.openSnackBar(error.error.message, 3000);
-    });
+    }, error => this.snackbar.openSnackBar(error.error.message, 4000));
+  }
+
+  getGroupUsers() {
+    let group = localStorage.getItem('groupName');
```

Comment:
nu am gasit nicaieri unde sa se seteze groupName in localStorage, in plus de ce se face?

Reply:
in user-groups.ts avem un button pentru groupUsers,aici puteam sa fiu mai specific si sa zic ca butonul e pentru groupDetails unde o sa afisam useri din grupul respective,in user-groups.ts  avem localStorage.setItem('groupName', group.name); pentru ca in group-users avem un api call care are nevoie de groupName.